src/core/qgsexpression.cpp

```diff
@@ -3156,6 +3156,143 @@ static QVariant fcnGetLayerProperty( const QVariantList& values, const QgsExpres
   return QVariant();
 }
 
+static QVariant fcnArray( const QVariantList& values, const QgsExpressionContext*, QgsExpression* )
+{
+  return values;
+}
+
+static QVariant fcnArrayLength( const QVariantList& values, const QgsExpressionContext*, QgsExpression* )
+{
+  return values.at( 0 ).toList().length();
```

Comment:
It'd be a good idea to protect these conversions with a similar function to the existing getIntValue/getDoubleValue/etc, and set the eval error string to notify users if they use a non-list value with these functions.